src/Platforms.jl

```diff
@@ -207,3 +207,55 @@ function Pkg.Artifacts.pack_platform!(meta::Dict, p::ExtendedPlatform)
     # override the default ones
     Artifacts.pack_platform!(meta, base_platform(p))
 end
+
+# Get the list of CPU features for the host system querying libLLVM
+function get_cpu_features()
+    if VERSION < v"1.4"
+        error("Cannot automatically detect the features of the CPU with this version of Julia!")
+    end
+    features = filter(f -> startswith(f, "+"),
+                      split(unsafe_string(ccall(:LLVMGetHostCPUFeatures, Cstring, ())),
+                            ","))
+    return sort!(replace.(features, r"^\+" => ""))
+end
+
+# Get the microarchitecture of a CPU from the list of its features.
+# TODO: support other architectures, like armv7l and aarch64.
+function march(cpu_features::Vector{String})
+    # List of CPU features from https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html
+    if all(in(cpu_features), ("mmx", "sse", "sse2")) # Generic x86_64
+        if all(in(cpu_features), ("sse3", "ssse3", "sse4.1", "sse4.2", "popcnt", "avx", "aes", "pclmul")) # avx
+            if all(in(cpu_features), ("movbe", "avx2", "fsgsbase", "rdrnd", "fma", "bmi", "bmi2", "f16c")) # avx2
+                # Some names are different: ADCX -> ADX, PREFETCHW -> PRFCHW
+                if all(in(cpu_features), ("pku", "rdseed", "adx", "prfchw", "clflushopt", "xsavec", "xsaves", "avx512f", "clwb", "avx512vl", "avx512bw", "avx512dq", "avx512cd")) # avx512
+                    return "avx512"
+                end
+                return "avx2"
+            end
+            return "avx"
+        end
+        return "x86_64"
+    else
+        @warn "Cannot determine the microarchitecture for the given set of features!"
+        return nothing
```

Comment:
I don't want a hard failure if we can't determine the microarchitecture from the given set of features, but it's better to issue a warning just in case we're we're getting strange results.  Also, this function would need to be used only when we're asking the set of features for an architecture we know we support (currently only `x86_64`), as done below in `extended_platform_key_abi`